### PR TITLE
fix[ux]: fix empty hints in error messages

### PR DESCRIPTION
**Pull request by sandbubbles**
### What I did
Remove empty hint message from exceptions. As in https://github.com/vyperlang/vyper/issues/4313.

### How I did it
Modify the check of absent hint to include an empty string.

### Commit message

```
remove empty hint message from exceptions. this regression was
introduced in 03095ce3921636. the root cause of the issue, however,
was that `hint=""` could be constructed in the first place. this commit
fixes the `get_levenshtein_error_suggestions` helper so that it returns
`None` on failure to find a suggestion rather than the empty string
`""`.
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by sandbubbles** [comment]
`UndeclaredException` is the only place where I encountered empty hints. An alternative solution is to add the check there to prevent the hint from being an empty string in the first place.

**Comment by charles-cooper** [comment]
It seems like a code smell that we are creating empty string hints in the first place. Do we know where they are coming from?

**Comment by sandbubbles** [comment]
> It seems like a code smell that we are creating empty string hints in the first place. Do we know where they are coming from?

I think its as in this -  https://github.com/vyperlang/vyper/pull/4351#issuecomment-2460349861 - just the `UndeclaredException`. But I can spend some more time if i encounter it elsewhere.

**Comment by charles-cooper** [comment]
note the regression comes from https://github.com/vyperlang/vyper/pull/4154 -- the previous code checked for truthy value of `self.hint`.

**Comment by sandbubbles** [comment]
> note the regression comes from #4154 -- the previous code checked for truthy value of `self.hint`.

Yeah, I think there are two main solutions:
1. Have the check in the same way as pre #4351 - (which is proposed in this PR, because i though hint can be only None or string, but I can rewrite it to `if not hint` ).
2. Or keep the current test `if hint is None` and modify `get_levenshtein_error_suggestions` to  return `None` instead of an empty string (which is used as the hint in one type of `UndeclaredDefinition`)

I think the second solution is cleaner, but the first solution ensures that if we missed any other empty string hints, they are covered by that.
What do you prefer?

**Comment by charles-cooper** [comment]
i think 2. is better -- we should not be supplying empty string as a hint!

**Comment by sandbubbles** [comment]
> i think 2. is better -- we should not be supplying empty string as a hint!

Hope its better now :)

File: vyper/semantics/analysis/levenshtein_utils.py [hunk 1]
```diff
@@ -99,12 +99,12 @@ def _get_levenshtein_error_suggestions(
         or an empty string.
     """
 
-    if key is None or key == "":
-        return ""
+    if not key:
```
Comment (charles-cooper) [inline vyper/semantics/analysis/levenshtein_utils.py:102]:
```suggestion
    if key is None or key == "":
```

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**